xbmc/filesystem/PluginDirectory.cpp

```diff
@@ -509,10 +511,10 @@ bool CPluginDirectory::WaitOnScriptResult(const std::string &scriptPath, int scr
     if (!m_fetchComplete.WaitMSec(20))
     {
       CScriptObserver scriptObs(scriptId, m_fetchComplete);
-      if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
-      {
+      CGUIWindowManager& wm = CServiceBroker::GetGUI()->GetWindowManager();
+      CGUIDialogProgress* progress = wm.IsModalDialogTopmost(WINDOW_DIALOG_PROGRESS) ? wm.GetWindow<CGUIDialogProgress>(WINDOW_DIALOG_PROGRESS) : nullptr;
```

Comment:
Small hint: the ternary operator usually leads to lines that are too long to fit on small screens. This can be avoided by doing something like:

```c++
CGUIDialogProgress* progress = nullptr;
if (wm.IsModalDialogTopmost(WINDOW_DIALOG_PROGRESS))
  progress = wm.GetWindow<CGUIDialogProgress>(WINDOW_DIALOG_PROGRESS);
```

Reply:
The ternary also makes this line too long. You could do something like:

```
if (progress != nullptr)
{
  if (!progress->WaitOnEvent(m_fetchComplete, 20))
    m_cancelled = true;
}
else if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
  m_cancelled = true;
```

It's usually better to add new lines than add complexity to existing lines. The most important thing is readability and maintainability for others.